framework/auth/core.py

```diff
@@ -843,6 +849,52 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def clean_email_verifications(self):
+        email_verifications = deepcopy(self.email_verifications)
+        for token in self.email_verifications:
+            try:
+                self.verify_token(token)
+            except (KeyError, ExpiredTokenError):
+                email_verifications.pop(token)
+        self.email_verifications = email_verifications
+        self.save()
```

Comment:
Avoid implicit saves in methods. The user of your method might want to do other updates before they save:

```python
user.clean_email_verifications()
user.do_something_else()
user.save()
```